docs/_posts/2021-06-30-luau-recap-june-2021.md

```diff
@@ -0,0 +1,58 @@
+---
+layout: single
+title:  "Luau Recap: June 2021"
+---
+
+Luau is our new language that you can read more about at [https://roblox.github.io/luau](https://roblox.github.io/luau). Most of our team was busy working on improving Luau interaction with Roblox Studio for an upcoming feature this month, but we were able to add typechecking and performance improvements as well!
+
+[Cross-posted to the [Roblox Developer Forum](https://devforum.roblox.com/t/luau-recap-june-2021/).]
+
+## Constraint Resolver
+
+To improve type inference under conditional expressions and other dynamic type changes (like assignments) we have introduced a new constraint resolver framework into Luau type checker.
```

Comment:
Add a simple example? Something like
```lua
function hi(name: string?)
  if (f == nil)
    print("Hello mysterious stranger!")
  else
   print("Hello %s!", name)
  end
end
```

Reply:
Yeah, I'll add something like this. Sadly, equality operator is a WIP, we have to use `if a` and `if not a` right now.